### DISCUSSION TITLE
Documenting Code Contribution

### DISCUSSION THREAD

**Discussion by liatgold** [cloud-custodian/cloud-custodian#7392] [General]
When contributing, is all of the documentation automatically taken care of with the tox build using the comments in the code itself? Or are there additional places where documentation needs to be added manually?

Additionally, could you provide instructions as to how to contribute to the source code page in the documentation (i.e. https://cloudcustodian.io/docs/_modules/c7n/resources/asg.html)? Or can that only be done once the contribution is approved and merged?

**Comment by kapilt** [comment]
see the developer docs on installation and documentation building here, https://cloudcustodian.io/docs/developer/documentation.html

tldr docs are built using sphinx, there are narrative docs in the docs folder, and there are example docs and schema extracted from source code for given resource / action that form up the reference material per the asg link you referenced.  doc changes are done through pull requests to this repo.

**Comment by kapilt** [comment]
beyond the existing dev docs there's some additional docs for aws resource additions in a recent pr  https://github.com/cloud-custodian/cloud-custodian/blob/84e77a1e8375bdd2919b320e4d5fe7a2ed26368c/docs/source/aws/contribute.rst
